pandas/core/algorithms.py

```diff
@@ -730,9 +746,22 @@ def factorize(
         codes, uniques = values.factorize(sort=sort)
         return _re_wrap_factorize(original, uniques, codes)
 
-    if not isinstance(values.dtype, np.dtype):
-        # i.e. ExtensionDtype
-        codes, uniques = values.factorize(na_sentinel=na_sentinel)
+    elif not isinstance(values.dtype, np.dtype):
+        if (
+            na_sentinel == -1
+            and "use_na_sentinel" in inspect.signature(values.factorize).parameters
```

Comment:
why are you inspecting like this? its *always* passed (with no_default maybe)

Reply:
We're inspecting whether the EA `values` can accept the new argument "use_na_sentinel". https://github.com/pandas-dev/pandas/pull/47157#discussion_r893098175

Reply:
Might be good to add a comment to that effect